### ISSUE TITLE
local package installation failing with pip version error

### ISSUE THREAD

**Issue by danielhoherd** [pipxproject/pipx#93]
This looks like it may be a dupe of #72 but since that is closed I'm opening a new ticket.

```
$ python3 --version
Python 3.7.2
$ pip3 freeze | grep pipx
pipx==0.12.1.0
$ pipx --version
0.12.1.0
$ pip2 --version
pip 19.0.2 from /usr/local/lib/python2.7/site-packages/pip (python 2.7)
$ pip3 --version
pip 19.0.2 from /usr/local/lib/python3.7/site-packages/pip (python 3.7)
$ pipx list
nothing has been installed with pipx 😴
$ pipx install -e plexdl
--upgrade should either be a path to a local project or a VCS url beginning with svn+, git+, hg+, or bzr+
You are using pip version 18.1, however version 19.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
'/Users/dho/.local/pipx/venvs/plexdl/bin/python -m pip install --editable --upgrade pip -q' failed
```

**Comment by r4lv** [comment]
When I used pipsi, I often did
``` bash
pipsi install -e .
```
which now translates to something like
``` bash
pipx install --spec="-e" --spec="." my-local-package
```

(note the `=` before the `"-e"`, which prevents the CLI engine to think `-e` is an argument)

Maybe something similar would work for you too?

**Comment by r4lv** [comment]
I think the problem is in these lines:
https://github.com/pipxproject/pipx/blob/f324bef820b7df2e78d9341788c08e2b858bf2dd/pipx/commands.py#L254-L256
where `pip_args` (which contain the `-e` from the command line) is passed to *both* the `pip install --upgrade pip` and the actual `pip install <your package>`.

That change was made 3 days ago in commit https://github.com/pipxproject/pipx/commit/f324bef820b7df2e78d9341788c08e2b858bf2dd.

@cs01 I think the `--editable` flag should only be passed to `venv.install_package`, and not to `venv.create_venv`. A possible solution would be to append the `--editable` flag to a new list `pip_install_args`, instead of:
https://github.com/pipxproject/pipx/blob/f324bef820b7df2e78d9341788c08e2b858bf2dd/pipx/main.py#L55

...or simply filter out the `--editable` from the `pip_args` here: https://github.com/pipxproject/pipx/blob/f324bef820b7df2e78d9341788c08e2b858bf2dd/pipx/Venv.py#L30

**Comment by cs01** [comment]
Sorry about this! @r4lv is right.  I fixed one problem but introduced another.  I’ll fix it soon (out of town atm though). In the mean time you can use an older version.

**Comment by cs01** [comment]
This has been fixed in 0.12.2.0 and a unit test was added to make sure the regression does not occur again :smile:
